src/classes/frDonorEmails.cls

```diff
@@ -10,7 +10,7 @@ public class frDonorEmails {
             email.subject = 'Funraise Email - ' + String.valueOf(request.get('subject'));
             email.MessageDate = DateTime.newInstance((Long)request.get('sentDate')).dateGMT();
             email.fr_Email_ID__c = funraiseId;
-            insert email;
+            Database.upsert(email, EmailMessage.Fields.fr_Email_ID__c, true);
```

Comment:
Did this so it's consistent with the way we insert records generally, using our fr_Id__c as a matcher key.  With this change though, there was a test that started to fail, but on review of the test it didn't make much sense anyway (we shouldn't ever be syncing from Funraise without an email id) so I removed the test.